shell/platform/darwin/macos/framework/Source/FlutterExternalTextureMetal.h

```diff
@@ -27,4 +27,15 @@
  */
 - (BOOL)populateTexture:(nonnull FlutterMetalExternalTexture*)metalTexture;
 
+/**
+ * Create FlutterMetalExternalTexture from YUV pixelbuffer.
+ */
+- (BOOL)populateTextureFromYUVAPixelBuffer:(nonnull FlutterMetalExternalTexture*)metalTexture
+                               pixelBuffer:(nonnull CVPixelBufferRef)pixelBuffer;
+/**
+ * Create FlutterMetalExternalTexture from RGBA pixelbuffer.
+ */
+- (BOOL)populateTextureFromRGBAPixelBuffer:(nonnull FlutterMetalExternalTexture*)metalTexture
+                               pixelBuffer:(nonnull CVPixelBufferRef)pixelBuffer;
```

Comment:
> 1. Why have duplicate methods that distinguish the pixel format by name, instead of one method that takes the pixel format as a parameter?

The reason for using two functions is to avoid `populateTexture` from being too long.  In `populateTexture` funciton has fellow code, so we already know the format before calling the function.
```
OSType pixel_format = CVPixelBufferGetPixelFormatType(pixelBuffer);
  if (pixel_format == kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange ||
      pixel_format == kCVPixelFormatType_420YpCbCr8BiPlanarFullRange) {
    return [self populateTextureFromYUVAPixelBuffer:textureOut pixelBuffer:pixelBuffer];
  } else {
    return [self populateTextureFromRGBAPixelBuffer:textureOut pixelBuffer:pixelBuffer];
  }
```

> 2. the first parameter is a texture, not a pixel buffer as indicated by the naming here

Thanks for pointing out the problem, I will modify it now.

Reply:
> 1. Why have duplicate methods that distinguish the pixel format by name, instead of one method that takes the pixel format as a parameter?

The reason for using two functions is to avoid `populateTexture` from being too long.  In `populateTexture` funciton has fellow code, so we already know the format before calling the function.
```
OSType pixel_format = CVPixelBufferGetPixelFormatType(pixelBuffer);
  if (pixel_format == kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange ||
      pixel_format == kCVPixelFormatType_420YpCbCr8BiPlanarFullRange) {
    return [self populateTextureFromYUVAPixelBuffer:textureOut pixelBuffer:pixelBuffer];
  } else {
    return [self populateTextureFromRGBAPixelBuffer:textureOut pixelBuffer:pixelBuffer];
  }
```

> 2. the first parameter is a texture, not a pixel buffer as indicated by the naming here

Thanks for pointing out the problem, I will modify it now.